domain/secretbackend/state/state.go

```diff
@@ -404,6 +387,53 @@ WHERE uuid = $M.uuid`, sqlair.M{}, SecretBackendRotationRow{})
 	return domain.CoerceError(err)
 }
 
+// SetModelSecretBackend sets the secret backend for the given model.
+func (s *State) SetModelSecretBackend(ctx context.Context, modelUUID coremodel.UUID, backendName string) error {
+	db, err := s.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	modelBackendUpsertStmt, err := s.Prepare(`
+INSERT INTO model_secret_backend
+	(model_uuid, secret_backend_uuid)
+VALUES ($M.model_uuid, $M.secret_backend_uuid)
+ON CONFLICT (model_uuid) DO UPDATE SET
+	secret_backend_uuid = $M.secret_backend_uuid`, sqlair.M{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		sb, err := s.getSecretBackend(ctx, tx, "name", backendName)
```

Comment:
We don't need this separate look up - we can do the insert with a sub select. 
And besides, looking up an hydrating a whole backend just to get the id is inefficient.

Reply:
We need this for checking existence because the sqlaire does not tell us the affected rows.